src/call-api.ts

```diff
@@ -0,0 +1,153 @@
+import * as abi from "@api3/airnode-abi";
+import * as adapter from "@api3/airnode-adapter";
+import * as node from "@api3/airnode-node";
+import * as ois from "@api3/airnode-ois";
+import * as ethers from "ethers";
+import isNil from "lodash/isNil";
+import { ApiValuesByBeaconId, RrpBeaconServerKeeperTrigger } from "./types";
+import { retryGo } from "./utils";
+
+export const readApiValue = async (
+  airnodeHDNode: ethers.utils.HDNode,
+  oises: ois.OIS[],
+  apiCredentials: node.ApiCredentials[],
+  {
+    oisTitle,
+    endpointName,
+    templateId,
+    templateParameters,
+    overrideParameters,
+  }: RrpBeaconServerKeeperTrigger
+): Promise<node.LogsData<ApiValuesByBeaconId>> => {
+  let apiValue: ethers.BigNumber | null = null;
```

Comment:
You probably saw this coming, but... now that you've moved all this into a function you can remove the let declaration. It's not a biggy of course.

The value of apiValue is always null, until after a successful call to the API in your try below, so all the other instances can be replaced with null and the successful response can be immediately returned.

```
  const encodedTemplateParameters = abi.encode(templateParameters);
  const expectedTemplateId = node.evm.templates.getExpectedTemplateId({
    airnodeAddress,
    endpointId,
    encodedParameters: encodedTemplateParameters,
    id: templateId,
  });
  if (expectedTemplateId !== templateId) {
    return [
      [
        node.logger.pend(
          "ERROR",
          `templateId '${templateId}' does not match expected templateId '${expectedTemplateId}'`
        ),
      ],
      { [beaconId]: null },
    ];
  }
```

```
  // Extract API value
  try {
    const response = adapter.extractAndEncodeResponse(
      apiResponse!.data,
      reservedParameters as adapter.ReservedParameters
    );
    const apiValue = ethers.BigNumber.from(response.values[0].toString());

    const logApiValue = node.logger.pend(
      "INFO",
      `API value: ${apiValue.toString()}`
    );

    return [[logApiResponse, logApiValue], { [beaconId]: apiValue }]
  } catch (error) {
    return [
      [
        node.logger.pend(
          "ERROR",
          `failed to extract or encode value from API response: ${JSON.stringify(
            apiResponse.data
          )}`,
          error as any
        ),
      ],
      { [beaconId]: null },
    ];
  }
```